src/system_data.jl

```diff
@@ -1142,6 +1142,34 @@ function get_available_associated_components(
     return components
 end
 
+"""
+Return a vector of NamedTuples with pairs of components and supplemental attributes that
+are associated with each other.
+"""
+function get_component_supplemental_attribute_pairs(
```

Comment:
It's not _quite_ as evident to me that we need a `get_available...` here as in your previous case https://github.com/NREL-Sienna/InfrastructureSystems.jl/pull/468/files#r2103334665, but seems like we should still probably have one? I could be convinced otherwise.